edit-trace/benches/main.rs

```diff
@@ -1,26 +1,44 @@
-use automerge::{Automerge, Value, ROOT};
+use automerge::{AutoTxn, Automerge, Value, ROOT};
 use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
 use std::fs;
 
-fn replay_trace(commands: Vec<(usize, usize, Vec<Value>)>) -> Automerge {
+fn replay_trace_tx(commands: Vec<(usize, usize, Vec<Value>)>) -> Automerge {
     let mut doc = Automerge::new();
+    let mut tx = doc.transaction();
+    let text = tx.set(&ROOT, "text", Value::text()).unwrap().unwrap();
+    for (pos, del, vals) in commands {
+        tx.splice(&text, pos, del, vals).unwrap();
+    }
+    tx.commit();
```

Comment:
Is this really a valid benchmark? This generates one commit right at the end of all the edits, which seems like a pretty unlikely workflow. On the other hand, it probably doesn't make sense to generate a transaction on every single change (although I'm sure there are users who will). Probably the most rigorous way to approach this is to parameterise this benchmark by some kind of transaction frequency and measure performance across all of them. Is this something we care about doing or should we just view this benchmark as somewhat synthetic?

Reply:
There is no reason to cut a transaction unless you are doing real time collaboration or saving the doc to disk.  So this (in my mind) represents someone editing offline and hitting "save" at the end of the edit session.

Reply:
Hmm. Yeah I think that makes sense, nice!